cloudpub/ms_azure/utils.py

```diff
@@ -306,12 +306,24 @@ def update_skus(
     # The alternate plan name ends with the suffix "-genX" and we can't change that once
     # the offer is live, otherwise it will raise "BadRequest" with the message:
     # "The property 'PlanId' is locked by a previous submission".
-    default_gen = "V2"
-    alt_gen = "V1"
     for osku in old_skus:
+        # Get the security type for all gens
         if osku.security_type is not None:
             security_type = osku.security_type
-        if osku.id.endswith("-gen2"):  # alternate is gen2 hence V1 is the default.
+
+        # Default Gen2 cases
+        if osku.image_type.endswith("Gen1") and osku.id.endswith("gen1"):
+            default_gen = "V2"
+            alt_gen = "V1"
+        elif osku.image_type.endswith("Gen2") and not osku.id.endswith("gen2"):
+            default_gen = "V2"
+            alt_gen = "V1"
+
+        # Default Gen1 cases
+        elif osku.image_type.endswith("Gen1") and not osku.id.endswith("gen1"):
+            default_gen = "V1"
+            alt_gen = "V2"
+        elif osku.image_type.endswith("Gen2") and osku.id.endswith("gen2"):
             default_gen = "V1"
             alt_gen = "V2"
```

Comment:
Do we need to iterate over all `old_skus` ?
Looks like checking the latest one might be enough.

Or maybe we just need tore unit tests :to cover the case with more entries in the old_skus

Reply:
Hmm you're right, we may just check the first one to determine the second one, no need to iterate.

Let me update the code and make sure the tests passes